routers/web/auth/auth.go

```diff
@@ -386,6 +386,13 @@ func getUserName(gothUser *goth.User) (string, error) {
 	switch setting.OAuth2Client.Username {
 	case setting.OAuth2UsernameEmail:
 		return user_model.NormalizeUserName(strings.Split(gothUser.Email, "@")[0])
+	case setting.OAuth2UsernamePreferredUsername:
+		preferredUsername, exists := gothUser.RawData["preferred_username"]
+		if exists {
+			return user_model.NormalizeUserName(preferredUsername.(string))
+		} else {
+			return "", fmt.Errorf("preferred_username is missing in received user data but configured as username source. Check if OPENID_CONNECT_SCOPES contains profile")
```

Comment:
Can we perhaps log the username that has no preferred username as well?

Reply:
Do you mean the gothUser.NickName?

Reply:
Yeah exactly, for example

Reply:
Maybe the userId would be the better option, because in most cases when the preferred_username is missing (in case the profile scope is missing), the nickname would be missing too, but the userId is already available if the openid scope is set and the openid scope is added by default.
But i don't know if the userId of the id-provider would be a helpful information.

Reply:
For debugging at least yes.
Anything is better than no information which user is problematic.

Reply:
I believe the error should already have been outputted.

`ServerError` does log the error in it.

![image](https://github.com/go-gitea/gitea/assets/2114189/859e7ab9-1eaf-4a5b-aef3-5aa3bc56e639)